tests/suites/test_suite_psa_crypto.function

```diff
@@ -4074,13 +4075,26 @@ void aead_encrypt( int key_type_arg, data_t *key_data,
     PSA_ASSERT( psa_import_key( &attributes, key_data->x, key_data->len,
                                 &key ) );
 
-    PSA_ASSERT( psa_aead_encrypt( key, alg,
-                                  nonce->x, nonce->len,
-                                  additional_data->x, additional_data->len,
-                                  input_data->x, input_data->len,
-                                  output_data, output_size,
-                                  &output_length ) );
+    status = psa_aead_encrypt( key, alg,
+                               nonce->x, nonce->len,
+                               additional_data->x, additional_data->len,
+                               input_data->x, input_data->len,
+                               output_data, output_size,
+                               &output_length );
 
+#if defined(MBEDTLS_AES_ALT) || \
```

Comment:
Well, this doesn't seem a scalable approach to me and I don't feel I can approve that. We have to find a better solution for the problem of the dependencies of tests going beyond the configuration options: key size, nonce size ...

Reply:
I don't think there's going to be a scalable solution before we get the driver JSON autogeneration done, and thus can autogenerate defines based on what's actually accelerator-supported in terms of key sizes. Nonce size I don't think is included in the driver JSON.

This PR is what we have had to do to enable testing on our targets. I agree that it is food for thought for a future improvement (based on the driver JSON), but for now, we'd highly appreciate getting this in. I feel the exception case for skipping is so well-constrained that it in practice won't have impact on anything else besides on-target testing with accelerators present (which you guys don't do yet?).

Reply:
Even with driver descriptions, there will always be cases of partial support that can't be expressed in the description language. So I think the logic here is the best we can do (possibly wrapped in some macros/functions if applicable): try the operation, require either success or not-supported, and the built-in implementation must succeed but drivers/ALT are allowed to return not-supported.

Reply:
Looking at this, I am wondering if we want to support the checks on MBEDTLS_AES_ALT and MBEDTLS_AES_SETKEY_ENC_ALT ? I guess this is for the case where the AES accelerator is leveraged through the Mbed TLS AES implementation. Don't we want to avoid that and that an AES accelerator is only leveraged through the PSA driver interface. @stevew817, is it currently an issue for you if we remove  #if defined(MBEDTLS_AES_ALT) ||  defined(MBEDTLS_AES_SETKEY_ENC_ALT) here?

Reply:
It does make sense to use ALT interfaces for accelerators where they exist, since that automatically makes them available under both PSA and non-PSA APIs. PSA is the future, but the present includes ALT as well.

Reply:
Yes, that would be an issue for us. We're not creating PSA drivers for our 'older' parts (which only have an AES accelerator), and instead are continuing to rely on the mbed TLS implementation and accelerating the primitive underneath there. That's the best we can do ATM to avoid re-implementing a bunch of sensitive software algorithms in a patchwork driver.

Not having those guards here would cause test case failure for those parts.

Reply:
And since @gilles-peskine-arm beat me to the punch while I was typing my comment: see his answer for the tl;dr.